src/ParquetFileViewer/FieldSelectionDialog.cs

```diff
@@ -93,17 +96,36 @@ private void FieldsToLoadForm_Load(object sender, EventArgs e)
                         if (!fieldNames.Contains(field.Name.ToLowerInvariant())) //Normally two fields with the same name shouldn't exist but lets make sure
                         {
                             bool isUnsupportedFieldType = field.SchemaType == SchemaType.List || field.SchemaType == SchemaType.Map || field.SchemaType == SchemaType.Struct;
-                            this.fieldsPanel.Controls.Add(
-                                new CheckBox()
+                            var fieldCheckbox = new CheckBox()
+                            {
+                                Name = string.Concat("checkbox_", field.Name),
+                                Text = string.Concat(field.Name, isUnsupportedFieldType ? "(Unsupported)" : string.Empty),
+                                Tag = field.Name,
+                                Checked = this.PreSelectedFields.Contains(field.Name),
+                                Location = new Point(locationX, locationY),
+                                AutoSize = true,
+                                Enabled = !isUnsupportedFieldType
+                            };
+                            fieldCheckbox.CheckedChanged += (object checkboxSender, EventArgs checkboxEventArgs) =>
+                            {
+                                var fieldCheckBox = (CheckBox)checkboxSender;
+                                if (!fieldCheckBox.Checked)
                                 {
-                                    Name = string.Concat("checkbox_", field.Name),
-                                    Text = string.Concat(field.Name, isUnsupportedFieldType ? "(Unsupported)" : string.Empty),
-                                    Tag = field.Name,
-                                    Checked = this.PreSelectedFields.Contains(field.Name),
-                                    Location = new Point(locationX, locationY),
-                                    AutoSize = true,
-                                    Enabled = !isUnsupportedFieldType
-                                });
+                                    foreach (Control control in this.fieldsPanel.Controls)
+                                    {
+                                        if (control.Tag.Equals(SelectAllCheckboxName) && control is CheckBox checkbox)
+                                        {
+                                            if (checkbox.Enabled && checkbox.Checked)
+                                            {
+                                                isClearingSelectAllCheckbox = true;
+                                                checkbox.Checked = false;
+                                                isClearingSelectAllCheckbox = false;
+                                            }

```

Comment:
Great addition, thanks! 

Not a big deal but maybe we could add a **break** statement at the end of the **if** here?

Reply:
Good point, added.